### PR TITLE
French translation completed

### PR DESCRIPTION
**Pull request by gege2b**
mainly all the UBL stuff

### CODE REVIEW

File: Marlin/language_fr.h [hunk 1]
```diff
@@ -201,26 +269,32 @@
 
 #define MSG_INFO_MENU                       _UxGT("Infos imprimante")
 #define MSG_INFO_PRINTER_MENU               _UxGT("Infos imprimante")
+#define MSG_3POINT_LEVELING                 _UxGT("Niveau à 3 points")
+#define MSG_LINEAR_LEVELING                 _UxGT("Niveau linéaire")
+#define MSG_BILINEAR_LEVELING               _UxGT("Niveau bilinéaire")
+#define MSG_UBL_LEVELING                    _UxGT("Niveau lit unifié (UBL)")
```
Comment (thinkyhead) [inline Marlin/language_fr.h]: 20 character limit for static LCD messages. And 18 is better.
This string will always be chopped off:
```
12345678901234567890
Niveau lit unifié (U
```

File: Marlin/language_fr.h [hunk 2]
```diff
@@ -133,6 +198,7 @@
 #define MSG_STORE_EEPROM                    _UxGT("Sauver config")
 #define MSG_LOAD_EEPROM                     _UxGT("Lire config")
 #define MSG_RESTORE_FAILSAFE                _UxGT("Restaurer défauts")
+#define MSG_INIT_EEPROM                     _UxGT("Initialisation EEPROM")
```
Comment (thinkyhead) [inline Marlin/language_fr.h]: This string will always be chopped off:
```
12345678901234567890
Initialisation EEPRO
```

File: Marlin/language_fr.h [hunk 3]
```diff
@@ -239,32 +313,32 @@
...
-  #define MSG_FILAMENT_CHANGE_RESUME_2        _UxGT("pour reprendre")
+#define MSG_FILAMENT_CHANGE_INIT_1          _UxGT("Attente Démarrage")
+#define MSG_FILAMENT_CHANGE_INIT_2          _UxGT("du filament")
+#define MSG_FILAMENT_CHANGE_INIT_3          _UxGT("changer")
+#define MSG_FILAMENT_CHANGE_UNLOAD_1        _UxGT("attente de")
+#define MSG_FILAMENT_CHANGE_UNLOAD_2        _UxGT("décharger filament")
+#define MSG_FILAMENT_CHANGE_INSERT_1        _UxGT("insérer filament")
+#define MSG_FILAMENT_CHANGE_INSERT_2        _UxGT("et app. bouton")
+#define MSG_FILAMENT_CHANGE_INSERT_3        _UxGT("pour continuer...")
+#define MSG_FILAMENT_CHANGE_HEAT_1          _UxGT("Appuyer sur le bouton...")
```
Comment (thinkyhead) [inline Marlin/language_fr.h]:
```
12345678901234567890
Appuyer sur le bouto
```

File: Marlin/language_fr.h [hunk 4]
```diff
@@ -239,32 +313,32 @@
...
+#define MSG_FILAMENT_CHANGE_HEAT_2          _UxGT("Pour chauffer la buse.")
```
Comment (thinkyhead) [inline Marlin/language_fr.h]:
```
12345678901234567890
Pour chauffer la bus
```

File: Marlin/language_fr.h [hunk 5]
```diff
@@ -74,6 +75,68 @@
...
+#define MSG_UBL_OUTPUT_MAP                  _UxGT("Voir maille")
+#define MSG_UBL_OUTPUT_MAP_HOST             _UxGT("Voir pour hôte")
+#define MSG_UBL_OUTPUT_MAP_CSV              _UxGT("Voir pour CSV")
+#define MSG_UBL_OUTPUT_MAP_BACKUP           _UxGT("Off Printer Backup")
+#define MSG_UBL_INFO_UBL                    _UxGT("Voir info UBL")
+#define MSG_UBL_EDIT_MESH_MENU              _UxGT("Modifier maille")
+#define MSG_UBL_FILLIN_AMOUNT               _UxGT("Taux de remplissage")
+#define MSG_UBL_MANUAL_FILLIN               _UxGT("Remplissage manuel")
+#define MSG_UBL_SMART_FILLIN                _UxGT("Remplissage auto")
+#define MSG_UBL_FILLIN_MESH                 _UxGT("Maille de remplissage")
```
Comment (thinkyhead) [inline Marlin/language_fr.h]:
```
12345678901234567890
Maille de remplissag
```

File: Marlin/language_fr.h [hunk 6]
```diff
@@ -74,6 +75,68 @@
...
+#define MSG_USER_MENU                       _UxGT("Commandes perso")
+
+#define MSG_UBL_DOING_G29                   _UxGT("G29 en cours")
+#define MSG_UBL_UNHOMED                     _UxGT("Origine XYZ d'abord")
+#define MSG_UBL_TOOLS                       _UxGT("Outils UBL")
+#define MSG_UBL_LEVEL_BED                   _UxGT("Niveau lit unifié (UBL)")
+#define MSG_UBL_MANUAL_MESH                 _UxGT("Maillage manuel")
+#define MSG_UBL_BC_INSERT                   _UxGT("Placer câle et mésurer")
+#define MSG_UBL_BC_INSERT2                  _UxGT("Mesure")
+#define MSG_UBL_BC_REMOVE                   _UxGT("Enlever et mesurer lit")
```
Comment (thinkyhead) [inline Marlin/language_fr.h]:
```
12345678901234567890
Enlever et mesurer l
```

File: Marlin/language_fr.h [hunk 7]
```diff
@@ -74,6 +75,68 @@
...
 #define MSG_LEVEL_BED                       _UxGT("Règl. Niv. lit")
+#define MSG_EDITING_STOPPED                 _UxGT("Arrêt edit. maillage")
+#define MSG_USER_MENU                       _UxGT("Commandes perso")
+
+#define MSG_UBL_DOING_G29                   _UxGT("G29 en cours")
+#define MSG_UBL_UNHOMED                     _UxGT("Origine XYZ d'abord")
+#define MSG_UBL_TOOLS                       _UxGT("Outils UBL")
+#define MSG_UBL_LEVEL_BED                   _UxGT("Niveau lit unifié (UBL)")
+#define MSG_UBL_MANUAL_MESH                 _UxGT("Maillage manuel")
+#define MSG_UBL_BC_INSERT                   _UxGT("Placer câle et mésurer")
```
Comment (thinkyhead) [inline Marlin/language_fr.h]:
```
12345678901234567890
Placer câle et mésur
```

File: Marlin/language_fr.h [hunk 8]
```diff
@@ -74,6 +75,68 @@
...
 #define MSG_BED_LEVELING                    _UxGT("Règl. Niv. lit")
 #define MSG_LEVEL_BED                       _UxGT("Règl. Niv. lit")
+#define MSG_EDITING_STOPPED                 _UxGT("Arrêt edit. maillage")
+#define MSG_USER_MENU                       _UxGT("Commandes perso")
+
+#define MSG_UBL_DOING_G29                   _UxGT("G29 en cours")
+#define MSG_UBL_UNHOMED                     _UxGT("Origine XYZ d'abord")
+#define MSG_UBL_TOOLS                       _UxGT("Outils UBL")
+#define MSG_UBL_LEVEL_BED                   _UxGT("Niveau lit unifié (UBL)")
+#define MSG_UBL_MANUAL_MESH                 _UxGT("Maillage manuel")
```
Comment (thinkyhead) [inline Marlin/language_fr.h]:
```
12345678901234567890
Niveau lit unifié (U
```

Comment (thinkyhead) [reply Marlin/language_fr.h]:
```
12345678901234567890
Test barre progressi
```

**Comment by thinkyhead** [comment]
A few of these strings are too long. 20 character limit for static messages. 18 character limit for menu items.

**Comment by gege2b** [comment]
wow OK, some of them are here since a long time

note : the "(UBL)" part on some string is here to keep a reference to the common term used in english (for pple who want to search something about UBL).

It's not strictly necessary even if it's helpfull

**Comment by gege2b** [comment]
here we go, strings snipped (not always easy)

just for the record, do you have a tool/script to measure the length of strings directly inside the language files ?

**Comment by thinkyhead** [comment]
Ready to merge at #7118.

**Pull request closed without merging**